# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1392348584 18000
#      Thu Feb 13 22:29:44 2014 -0500
# Node ID 00ba456b9aa82f4e8e189310b3e2af82c940c10e
# Parent  3afae1432cd67b5ba3fe8bb0c1852c0dd3a35c7b
Fix dispatch class of property accessor methods.

* ov-classdef (cdef_class::make_meta_class): Make sure the property
accessor methods have the correct dispatch class.

diff --git a/libinterp/octave-value/ov-classdef.cc b/libinterp/octave-value/ov-classdef.cc
--- a/libinterp/octave-value/ov-classdef.cc
+++ b/libinterp/octave-value/ov-classdef.cc
@@ -2933,25 +2933,27 @@ cdef_class::make_meta_class (tree_classd
                   // methods from the temporary storage map, so we can detect which
                   // ones are invalid and do not correspond to a defined property.
 
                   std::map<std::string, octave_value>::iterator git =
                     get_methods.find (prop_name);
 
                   if (git != get_methods.end ())
                     {
+                      make_function_of_class (retval, git->second);
                       prop.put ("GetMethod", git->second);
                       get_methods.erase (git);
                     }
 
                   std::map<std::string, octave_value>::iterator sit =
                     set_methods.find (prop_name);
 
                   if (sit != set_methods.end ())
                     {
+                      make_function_of_class (retval, sit->second);
                       prop.put ("SetMethod", sit->second);
                       set_methods.erase (sit);
                     }
 
                   retval.install_property (prop);
                 }
             }
         }
